### PR TITLE
refactor(*): Move artifactory integration into module

### PR DESCRIPTION
**Pull request by robzienert**
This PR starts the effort of making Igor build system integrations built as
plugins. The first step is to break the integrations into modules so we have
a better sense of what the right abstractions are.

I bit this effort off purely to get more experience with real-world plugin
implementations that aren't stages. Figured Igor needed some organizational
love anyway, and since it's such a small service, it could be completed
reasonably quickly (if no plugin bugs arise :sweat_smile:)

Added all the people who have touched files I mucked around with.

### CODE REVIEW

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildContent.groovy [hunk 1]
```diff
@@ -18,6 +18,9 @@ package com.netflix.spinnaker.igor.history.model
 
 import com.netflix.spinnaker.igor.build.model.GenericProject
 
+/**
+ * TODO(rz): Cannot move to kork-core due to Jenkins dependency
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildContent.groovy]: Should be moved to kork-core or igor-core

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy [hunk 2]
```diff
@@ -16,6 +16,9 @@
 
 package com.netflix.spinnaker.igor.history.model
 
+/**
+ * TODO(rz): Cannot move to kork-core due to Jenkins dependency
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy]: Should also be moved

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy [hunk 3]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright 2016 Schibsted ASA.
+ * Copyright 2016 Schibsted, Inc.
```
Comment (jervi) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy]: I think our legal name is actually `Schibsted ASA`, not `Schibsted, Inc`, as that name implies that the company is registered in the U.S., right?

File: igor-monitor-artifactory/src/test/groovy/com/netflix/spinnaker/igor/artifactory/ArtifactoryBuildMonitorSpec.groovy [hunk 4]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright 2019 Pivotal, Inc.
+ * Copyright 2019 Netflix, Inc.
```
[the remark this answers is not in the document]
Comment (robzienert) [reply igor-monitor-artifactory/src/test/groovy/com/netflix/spinnaker/igor/artifactory/ArtifactoryBuildMonitorSpec.groovy]: Yes, intellij steamrolled a bunch of stuff but I guess one slipped through the cracks. :(

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildContent.groovy (hunk 1 above)
Comment (robzienert) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildContent.groovy]: 100% on all the movement comments, but I didn't want to do much refactoring to support moving the files for this PR. There would be a bunch of other PRs that follow this to refactor things beyond just code shifting.

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy (hunk 3 above)
Comment (robzienert) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/history/model/GenericBuildEvent.groovy]: Oops, yeah - I buggered things - I'll fix. :)

**Pull request merged**